plugins/woocommerce/tests/api-core-tests/README.md

```diff
@@ -127,25 +129,279 @@ After you run a test, it's best to restart the environment to start from a fresh
+				code: testCoupon.code,
+				amount: Number( coupon.amount ).toFixed( 2 ),
+				discount_type: coupon.discount_type,
+			} )
+		);
+```
+
+Verify [field names](https://github.com/woocommerce/woocommerce/blob/4ac1d822ac9082102536b0f7aa9cb0553965adaa/plugins/woocommerce/tests/api-core-tests/tests/coupons/coupons.test.js#L142) 
+```js
+			expect( id ).toBeDefined();
+			expect( code ).toEqual( expectedCouponCode );
+```
+
+Verify [field types](https://github.com/woocommerce/woocommerce/blob/d19c20491e5a7ade64c8fd530f01e0f3f3f7e29c/plugins/woocommerce/tests/api-core-tests/tests/shipping/shipping-method.test.js#L85)
+```js
+expect(typeof responseJSON.id).toEqual('string');
+```
+
+Verify [field values](https://github.com/woocommerce/woocommerce/blob/4ac1d822ac9082102536b0f7aa9cb0553965adaa/plugins/woocommerce/tests/api-core-tests/tests/coupons/coupons.test.js#L186) 
+```js
+expect( updatedCoupons[ 1 ].amount ).toEqual( '25.00' );
+```
+
+Verify [field length](https://github.com/woocommerce/woocommerce/blob/4ac1d822ac9082102536b0f7aa9cb0553965adaa/plugins/woocommerce/tests/api-core-tests/tests/coupons/coupons.test.js#L136)
+```js
+expect( actualCoupons ).toHaveLength( expectedCoupons.length );
+```
+
+Verify search [headers](https://github.com/woocommerce/woocommerce/blob/d19c20491e5a7ade64c8fd530f01e0f3f3f7e29c/plugins/woocommerce/tests/api-core-tests/tests/orders/orders.test.js#L2703)
+```js
+// Verify total page count.
+			expect( page1.headers()[ 'x-wp-total' ] ).toEqual(
+				ORDERS_COUNT.toString()
+			);
+			expect( page1.headers()[ 'x-wp-totalpages' ] ).toEqual( '3' );
+  ```
+
+Verify Variable [not undefined](https://github.com/woocommerce/woocommerce/blob/778cb130f27d0dd0dc7da1acb0e89762f81c0f18/plugins/woocommerce/tests/api-core-tests/tests/coupons/coupons.test.js#L31)
+```js
+expect( couponId ).toBeDefined();
+```
+
+Verify [response contains an object](https://github.com/woocommerce/woocommerce/blob/778cb130f27d0dd0dc7da1acb0e89762f81c0f18/plugins/woocommerce/tests/api-core-tests/tests/coupons/coupons.test.js#L73)
+```js
+expect( await response.json() ).toEqual(
+			expect.objectContaining( updatedCouponDetails )
+		);
+```
+
+Verify [response contains an array containing an object](https://github.com/woocommerce/woocommerce/blob/778cb130f27d0dd0dc7da1acb0e89762f81c0f18/plugins/woocommerce/tests/api-core-tests/tests/coupons/coupons.test.js#L388)
+```js
+		expect( responseJSON.coupon_lines[ 0 ].meta_data ).toEqual(
+			expect.arrayContaining( [
+				expect.objectContaining( {
+					key: 'coupon_data',
+					value: expect.objectContaining( {
+						code: testCoupon.code,
+					} ),
+				} ),
+			] )
+		);
 ```
 
 ## Debugging tests
 
-For Playwright debugging, follow [Playwright's documentation](https://playwright.dev/docs/debug).
+When debugging API tests, it is usually simple enough to add `console.log()` statements to output the values of your response/JSON/variables/status etc. Be sure to remove them when done ;) 
```

Comment:
I noticed we're removing the reference to Playwright's debugging, does the debugging provided by Playwright not work well for API tests?

Reply:
Great question - The debugging provided by Playwright uses Playwright Inspector (a GUI tool) which operates only when there are browser interactions so it isn't applicable for API tests.
e.g if you were debugging one of the e2e tests you would make a call such as `npx playwright test example --debug` or pnpm equivalent which would fire up the inspector 
This wouldn't have any effect with the API tests. With that being said, I will remove the line mentioning `pnpm test:api-pw --debug (debug)` currently in the guide.

I also did a spot of googling and noticed a [feature request](https://github.com/microsoft/playwright/issues/19177) to to make debugging easier for API tests (added last week) - I have subscribed and if that comes to the fore then I will update accordingly

Reply:
> I also did a spot of googling and noticed a https://github.com/microsoft/playwright/issues/19177 to to make debugging easier for API tests

Nice! That would be cool to have that feature added in.

Debating if it's worth noting that the Playwright debugger won't work for the API tests, so for now it's easier/best to use `console.log()` statements to debug? Additionally, I think we could recommend using a plugin to see the API requests within WordPress, something like [this plugin](https://wordpress.org/plugins/wp-rest-api-log/)?

Reply:
Fully agree with this - the comment should be updated for clarity and informational purposes - good shout 👍

Also - great suggestion on the plugin - I'll take a peek and see how it looks 🙌🏻

Reply:
Thank you! The additions look good to me, thanks for adding this in!